### PR TITLE
Multiple data backends

### PR DESCRIPTION
**Pull request by ilmarinen**
Abstract away the SQLite specific data interface behind a general DataLayer api.
Allow the SQLite specific layer to be selected at runtime, in preparation for adding additional data layers.

### CODE REVIEW

File: endpoint/data_backends/__init__.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import importlib
+
+
+class DataLayer(object):
+    api = None
```
Comment (dpflann) [inline endpoint/data_backends/__init__.py:5]: What do you think of this?

Can the ‘api’ be encloses by the DataLayer?
Such that method calls on the ‘api’ are internalized?
Like ‘DataLayer.init(...)’ is a wrapper for ‘DataLayer.api.init(...)’.

File: endpoint/data_backends/sqlite/manage.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+
+def init_db():
+    db.create_all()
+
+
+def get_user_by_username(username):
+    user = User.query.filter(User.username == username).first()
+    if not user:
+        return False
```
Comment (dpflann) [inline endpoint/data_backends/sqlite/manage.py]: What do you think of this?

What is object returned by the query? Right now I am thinking of how returning different types (user and Boolean) could be confusing. Can you return ‘None’??

File: endpoint/packages/user/__init__.py [hunk 3]
```diff
@@ -1,23 +1,20 @@
...
+from endpoint.data_backends import DataLayer
 
 
 login_manager = LoginManager()
 
 
-def init(app, admin):
+def init(app):
+    # DataLayer.load_api(app.config["DATASTORE"])
+    # DataLayer.api.init(app)
```
Comment (dpflann) [inline endpoint/packages/user/__init__.py]: Are these commented out for debugging or should they deleted or uncommented?

Comment (ilmarinen) [reply endpoint/packages/user/__init__.py]: That is dead code, will remove. Thanks!

File: endpoint/data_backends/sqlite/manage.py (hunk 2 above)
Comment (ilmarinen) [reply endpoint/data_backends/sqlite/manage.py]: Good point, I was thinking about this too and how we deal with this at my day job. What we do is that we have a custom but very thin `Results` object wrapper that has a success, data and errors attribute. If the result is successful then `results.success` is set to `True` and the `results.data` is set to a dict that contains everything you want to return. If the call fails for some reason then `results.success` is set to `False` and `results.errors` is set to a list of exceptions or text string representations of the errors. Do you think I should do that here? Totally could and would be a nice consistent way to move forward.

File: endpoint/data_backends/__init__.py (hunk 1 above)
Comment (ilmarinen) [reply endpoint/data_backends/__init__.py:5]: Yup yup ... great suggestion. Thanks, will do this!

Comment (dpflann) [reply endpoint/data_backends/__init__.py:5]: I like that! 👌👍. We can extend the object if necessary.

**Comment by dpflann** [comment]
I think the code is good to go. Can you add information to the `README` about the config file format and naming format?

**Comment by ilmarinen** [comment]
@dpflann bump

**Review by dpflann — APPROVED** [review]

**Pull request merged**